From c61b5ccfd0719946f838e2c57595cf6c6d6cdb4b Mon Sep 17 00:00:00 2001
From: "L. E. Segovia" <amy@centricular.com>
Date: Wed, 23 Aug 2023 11:35:19 -0300
Subject: [PATCH 2/7] utils: Check that help2man can run if found

Meson doesn't check Perl interpreters for consistency, leading to
modules being not found only at run time.
---
 utils/meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/utils/meson.build b/utils/meson.build
index 5072c1cc..955d8bfc 100644
--- a/utils/meson.build
+++ b/utils/meson.build
@@ -42,7 +42,8 @@ pango_view = executable('pango-view', pango_view_sources,
 help2man = find_program('help2man', required: false)
 # help2man has to run pango-view to generate the man page, which it can't do if
 # we're cross building. We also can't tell help2man what exe wrapper to use.
-if help2man.found() and not meson.is_cross_build()
+
+if help2man.found() and run_command(help2man, '--version', check: false).returncode() == 0 and not meson.is_cross_build()
   help2man_opts = [
     '--no-info',
     '--section=1',
-- 
2.46.0

